sympy/physics/units/quantities.py

```diff
@@ -88,6 +88,10 @@ def _eval_is_positive(self):
     def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
+    def _eval_Abs(self):
+        return self.__class__(self.name, self.dimension,
```

Comment:
It doesn't look very Pythonic if I want to return new instance of the same class to use `self.func` instead of `self.__class__`. Can you please explain me, why the standard way is not the way to go?